storage/index.go

```diff
@@ -38,9 +38,46 @@ func loadSearchIndexAll(ctx context.Context, storageClient *storage.Client, buck
 	defer objectReader.Close()
 
 	var sia searchIndexAll
-	err = json.NewDecoder(objectReader).Decode(&sia)
-	if err != nil {
-		return nil, errors.Wrapf(err, "can't decode the index file (path: %s)", rootedIndexStoragePath)
+	dec := json.NewDecoder(objectReader)
+
+	for dec.More() {
```

Comment:
nit: Maybe leave a comment about why are you proposing this as it won't be super clear. We could simplify the same logic with `json.Unmarshal`.

Reply:
Added.